_posts/2022-01-24-data-prepper-1.2.1-performance-testing.md

```diff
@@ -0,0 +1,98 @@
+
+Throughout the test, Data Prepper can consistently maintain a higher throughput.
+
+![Response Time Graph](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Graph-Response-Time.png){: .img-fluid}
+
+Logstash hit peak latency of 7382ms
+Data Prepper’s peak latency was 5276ms
+
+|                                            | Data Prepper 1.2.1 | Logstash 7.13.2 |
+|:-------------------------------------------|-------------------:|----------------:|
+| Throughput                                 |           728 MB/s |        387 MB/s |
+| Mean response _(milliseconds)_             |              53 ms |           99 ms |
+| Total logs processed within 30 minute test |         68,166,000 |      36,206,800 |
+
+Data Prepper 1.2.1 processed 88% more logs than Logstash 7.13.2 within the 30-minute test limit.
+
+# Performance testing setup
+
+## Data Prepper Environment
+
+![Data Prepper Environment](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Data-Prepper.png){: .img-fluid}
+
+## Logstash Environment
+
+![Logstash Environment](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Logstash.png){: .img-fluid}
+
+Let’s compare the performance of the latest release of Data Prepper 1.2.1 against Logstash 7.13.2. We configured 10 clients to send requests as frequently as possible. Each request was 19.2 KB and contained a batch of 200 logs. We then ran this test for 30 minutes and measured the latency and throughput.
+
+## Application Configurations
+
+| Property             | Data Prepper | Logstash |
+| -------------------- | -----------: | -------: |
+| HTTP Source Thread   |            4 |        4 |
+| Max Connection Count |        2,000 |    n/a * |
+| Request Timeout      |       10,000 |    n/a * |
+| Buffer Size          |    2,000,000 |    n/a * |
+| Batch Size           |        5,000 |    5,000 |
+| Worker Thread        |           12 |       12 |
+| SSL                  |     Disabled | Disabled |
+
+_[*] Note, Max Connection Count, Request Timeout, Buffer Size are not configurable with Logstash 7.13.2_
+
+## Data Prepper Scaling
+
+In this simulation we ramp up from 1 client to 60 over one hour to measure the impact concurrent clients have on performance.
+
+![Data Prepper Throughput Increasing Users](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Graph-Throughput-Increasing-Users.png){: .img-fluid}
+
+As the number of clients increases Data Preppers throughput remains constant, processing an average of 10,473 MB/m.
+
+# Limitation of Testing
+
+It’s important to note that Data Prepper 1.2.1 and Logstash 7.13.2 support different feature sets, and our performance test suite is targeted at common functionality. As Data Prepper adds more processors and sources, we will expand our to cover those use cases, and we will continue update the community with performance benchmarks.
```

Comment:
- ambiguous plural pronouns ('our', 'we')
- You will probably change this when rephrasing the pronouns but this doesn't make sense: 'we will expand our to cover those use cases' (ours?)